e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/filter-payment-details.spec.js

```diff
@@ -26,52 +28,693 @@ context(`users can filter payment details by facility id and payment reference a
+
+        pages.utilisationReportPage.paymentDetailsTab.errorSummaryErrors().should('have.length', 0);
+        pages.utilisationReportPage.paymentDetailsTab.filters.facilityIdError().should('not.exist');
+      });
+    });
+
+    describe('when the facility id filter matches no payments', () => {
+      it('should display an error message for no payments found and persist the inputted value', () => {
+        const unknownFacilityIdFilter = '99999999';
+
+        const feeRecord = FeeRecordEntityMockBuilder.forReport(utilisationReport).withId(1).withFacilityId('11111111').build();
+        const payment = PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP).withFeeRecords([feeRecord]).build();
+
+        cy.task(NODE_TASKS.INSERT_FEE_RECORDS_INTO_DB, [feeRecord]);
+        cy.task(NODE_TASKS.INSERT_PAYMENTS_INTO_DB, [payment]);
+
+        pages.landingPage.visit();
+        cy.login(USERS.PDC_RECONCILE);
+
+        cy.visit(`/utilisation-reports/${reportId}`);
+
+        pages.utilisationReportPage.paymentDetailsTabLink().click();
+
+        cy.keyboardInput(pages.utilisationReportPage.paymentDetailsTab.filters.facilityIdInput(), unknownFacilityIdFilter);
+        pages.utilisationReportPage.paymentDetailsTab.filters.submitButton().click();
+
+        cy.url().should(
+          'eq',
+          relative(`/utilisation-reports/${reportId}?paymentDetailsPaymentReference=&paymentDetailsFacilityId=${unknownFacilityIdFilter}#payment-details`),
+        );
+
+        pages.utilisationReportPage.paymentDetailsTab.filters.panel().should('exist');
+
+        pages.utilisationReportPage.paymentDetailsTab.filters.facilityIdInput().should('have.value', unknownFacilityIdFilter);
+
+        pages.utilisationReportPage.paymentDetailsTab.filters.noRecordsMatchingFiltersText().should('exist');
+
+        pages.utilisationReportPage.paymentDetailsTab.paymentDetailsTable.row(payment.id, feeRecord.id).should('not.exist');
+      });
+    });
+  });
+
+  describe('payment currency filter', () => {
+    const assertAllPaymentCurrencyInputsAreNotChecked = () => {
+      Object.values(CURRENCY).forEach((currency) => {
+        pages.utilisationReportPage.paymentDetailsTab.filters.paymentCurrencyRadioInput(currency).should('not.be.checked');
+      });
+    };
+
+    describe('when a payment currency filter is submitted', () => {
+      it('should only display the payment with the supplied payment currency and persist the checked currency', () => {
+        const paymentCurrencyFilter = CURRENCY.USD;
+
+        const firstFeeRecord = FeeRecordEntityMockBuilder.forReport(utilisationReport).withId(1).withFacilityId('11111111').build();
+        const secondFeeRecord = FeeRecordEntityMockBuilder.forReport(utilisationReport).withId(2).withFacilityId('22222222').build();
+
+        const firstPayment = PaymentEntityMockBuilder.forCurrency(paymentCurrencyFilter)
```

Comment:
Seems to be a lot of duplication with the entity builders, are we able to make it more DRY?

Reply:
Simplified these a lot thanks, please see the new spec tests